inausoft.netCLI.all/inausoft.netCLI.Tests/RootCommandHandlerTests.cs

```diff
@@ -93,6 +97,89 @@ public void TestMethod2()
             Assert.AreEqual(0, mockCommand1Handler.LastRunParameters.IntOption);
         }
 
+        [TestMethod]
+        public void Command2WithoutParameters_ReturnsCorrectCommand()
+        {
+            //Arrange
+            var args = new string[]
+                            {
+                                "command2",
+                            };
+
+            //Act
+            var mockCommand1Handler = new MockCommand1Handler();
+            var mockCommand2Handler = new MockCommand2Handler();
+            var rootCommandHandler = new RootCommandHandler(new ICommandHandler[] { mockCommand2Handler, mockCommand1Handler });
+
+            var result = rootCommandHandler.Run(args);
+
+            //Assert
+            Assert.AreEqual(0, result);
+
+            Assert.IsNotNull(mockCommand2Handler.LastRunParameters);
+        }
+
+        [ExpectedException(typeof(InvalidCommandException))]
+        [TestMethod]
+        public void Command3WithoutRequiredParameter_ThrowsExcception()
```

Comment:
I would ommit this scenario - as it can be achieved on the handler level like:
```
class MockCommand3Handler : CommandHandler<Command3>
    {
        public Command3 LastRunParameters { get; private set; }

        public override int Run(Command3 command)
        {
            if(command.RequiredOption == null) { throw new SomeException(...) }

            LastRunParameters = command;

            return 0;
        }
    }
```

Reply:
Agreed